docs/tutorial/expressjs.rst

```diff
@@ -0,0 +1,430 @@
+    :start-after: [docs:ls-bare-rock]
+    :end-before: [docs:ls-bare-rock-end]
+    :dedent: 2
+
+We'll verify that the new ExpressJS rock is now approximately **15% smaller**
+in size! And that's just because of the simple change of ``base``.
+
+And the functionality is still the same. As before, we can confirm this by
+running the rock with Docker
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-chisel]
+    :end-before: [docs:docker-run-chisel-end]
+    :dedent: 2
+
+and then using the same ``curl`` request:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:curl-expressjs-bare-rock]
+    :end-before: [docs:curl-expressjs-bare-rock-end]
+    :dedent: 2
+
+The ExpressJS application should still respond with
+``Welcome to Express`` HTML.
+
+Cleanup
+~~~~~~~
+
+And that's it. We can now stop the container and remove the corresponding
+image:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker-chisel]
+    :end-before: [docs:stop-docker-chisel-end]
+    :dedent: 2
+
+.. _update-expressjs-application:
+
+Update the ExpressJS application
+================================
+
+As a final step, let's update our application. For example,
+we want to add a new ``/time`` endpoint which returns the current time.
+
+Start by creating the ``app/routes/time.js`` file in a text editor and paste the
+code from the snippet below:
+
+.. literalinclude:: code/expressjs/time.js
+    :language: javascript
+
+Place the code snippet below in ``app.js`` under routes registration section.
+It will register the new ``/time`` endpoint:
+
+.. literalinclude:: code/expressjs/time_app.js
+    :language: javascript
+    :start-after: [docs:append-lines]
+    :end-before: [docs:append-lines-end]
+
+Since we are creating a new version of the application, open the
+``rockcraft.yaml`` file and change the ``version`` (e.g. to ``0.2``).
+
+.. note::
+
+    ``rockcraft pack`` will create a new image with the updated code even if we
+    don't change the version. It is recommended to change the version whenever
+    we make changes to the application in the image.
+
+Pack and run the rock using similar commands as before:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:docker-run-update]
+    :end-before: [docs:docker-run-update-end]
+    :dedent: 2
+
+.. note::
+
+    Note that the resulting ``.rock`` file will now be named differently, as
+    its new version will be part of the filename.
+
+Finally, use ``curl`` to send a request to the ``/time`` endpoint:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:curl-time]
+    :end-before: [docs:curl-time-end]
+    :dedent: 2
+
+The updated application should respond with the current date and time (e.g.
+``Fri Jan 10 2025 03:11:44 GMT+0000 (Coordinated Universal Time)``).
+
+.. note::
```

Comment:
```suggestion
.. tip::
```

Reply:
This also isn't aligned with other documentation for the extensions, shall I change all the others to match the styles?

Reply:
No need – the rest of the team can update them later.